lib/util/exec/docker/index.ts

```diff
@@ -1,49 +1,85 @@
-import { uniq } from 'lodash';
-
 type Opt<T> = T | null | undefined;
 
+type VolumesPair = [string, string];
+export type VolumeOption = Opt<string> | Opt<VolumesPair>;
+
 export interface DockerOptions {
   image: string;
+  tag?: Opt<string>;
   dockerUser?: Opt<string>;
-  volumes?: Opt<Opt<string>[]>;
+  volumes?: Opt<VolumeOption[]>;
   envVars?: Opt<Opt<string>[]>;
   cwd?: Opt<string>;
-  tag?: Opt<string>;
-  preCommands?: Opt<string[]>;
-  postCommands?: Opt<string[]>;
+  preCommands?: Opt<Opt<string>[]>;
+  postCommands?: Opt<Opt<string>[]>;
```

Comment:
I have been wondering for the past week if we should actually just have `commands` instead of `preCommand` + `cmd` + `postCommands`. I maybe made it overly complex the first time, what do you think?

Reply:
It cuts two unnecessary terms (precommands and postcommands), but will lead to wrapping code `commands.push`-es around `commands.push(cmd)`.
I aimed to make it possible to write this way: 
```js
exec(cmd, { docker: dockerOptions(config) });
```
, where `dockerOptions()` return nullable and usually so small we can instead write:
```js
let docker;
if (config.binarySource === 'docker') docker = ...
```

Reply:
I think it only saves space to keep pre/post commands if it is "reusable" within a single manager?